src/components/structures/TimelinePanel.tsx

```diff
@@ -1058,7 +1058,7 @@ class TimelinePanel extends React.Component<IProps, IState> {
             } else {
                 cli.setRoomReadMarkers(
                     roomId,
-                    this.state.readMarkerEventId ?? "",
+                    this.state.readMarkerEventId ?? mkRoomMemberJoinEvent(cli.credentials.userId,roomId),
```

Comment:
The room member join event has already been created by synapse when the member joined.
`readMarkerEventId` is of type `string | null` but the room member join is a `MatrixEvent`.

Could you clarify what you're trying to get to?

Reply:
Read markers send an empty string as fully_read as defined in issue #23458, additionally according to the discussion, fully_read is no longer required for /read_markers since spec release v1.4, so on updated servers we can just leave out the field. Alternatively, just set it to the room join event by default instead of an empty string. I was trying to do so, as `joinRoom` event is returning  `Promise<Unkown>`, so it can't be used. I thought we can use this `mkRoomMemberJoinEvent` event.  But, ah, I think I saw this wrong and got this wrong, `MatrixEvent` return type won't do the thing.  
If I am getting it right, I think, `this.state.readMarkerEventId ?? mkRoomMemberJoinEvent(cli.credentials.userId,roomId).getId()` 
will work fine.

Reply:
Oh, I see, So If I am getting it right, `joinRoom` can't be a solution as we have already joined the room and we want to get the read marker ID, so it won't be any use, and the same in the case of `mkRoomMemberJoinEvent`, But if that's the case and taking a note of statement mentioned in the issue. I think we can assign **_null_** to `readMarkerEventId`. Please correct me if I'm wrong 😅

Reply:
And yes, Thanks for insight @t3chguy 😃